From c2a2435a619ddbe4a2c68f7664701e44e8571b96 Mon Sep 17 00:00:00 2001
From: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Date: Wed, 7 Mar 2018 11:45:22 +0200
Subject: [PATCH 3444/5242] Sound: SoC: codecs: Put AK4458 codec in manual
 mode

commit  4949a830c4489728cbe26c202b813a7be510c0e2 from
https://source.codeaurora.org/external/imx/linux-imx.git

We cannot both derive SAI BCLK for 384KHz-S32/768KHz-S16 and
respect the codec MCLK restrictions shown in AK4458 datasheet
Table 5, 6 and 7.
Since we can have same master clock for SAI and Codec in Manual
Mode, we've chosen to use it instead of Auto Mode.

Signed-off-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/ak4458.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/codecs/ak4458.h b/sound/soc/codecs/ak4458.h
index f906215..15ce6b0 100644
--- a/sound/soc/codecs/ak4458.h
+++ b/sound/soc/codecs/ak4458.h
@@ -12,6 +12,7 @@
 #include <linux/regmap.h>
 
 /* Settings */
+#define AK4458_ACKS_USE_MANUAL_MODE
 
 #define AK4458_00_CONTROL1			0x00
 #define AK4458_01_CONTROL2			0x01
-- 
1.7.9.5

